### PR TITLE
Update code based on PHPStan report

### PR DESCRIPTION
**Pull request by lukosius**

### CODE REVIEW

File: src/Yasumi/Provider/ChristianHolidays.php [hunk 1]
```diff
@@ -732,7 +732,7 @@ protected function calculateEaster($year, $timezone)
         if (extension_loaded('calendar')) {
             $easter_days = \easter_days($year);
         } else {
-            $golden = (int)(($year % 19) + 1); // The Golden Number
+            $golden = ($year % 19) + 1; // The Golden Number
```
[the remark this answers is not in the document]
Comment (lukosius) [reply src/Yasumi/Provider/ChristianHolidays.php]: Modulo will always return an integer, so the typecast is not needed. PHPStan also find this.

File: src/Yasumi/Provider/Japan.php [hunk 2]
```diff
@@ -440,14 +440,12 @@ private function calculateSubstituteHolidays()
                 }
 
                 // Add a new holiday that is substituting the original holiday
-                if (null !== $substituteDay) {
-                    $substituteHoliday = new Holiday('substituteHoliday:' . $shortName, [
-                        'en_US' => $date->translations['en_US'] . ' Observed',
-                        'ja_JP' => '振替休日 (' . $date->translations['ja_JP'] . ')',
-                    ], $substituteDay, $this->locale);
+                $substituteHoliday = new Holiday('substituteHoliday:' . $shortName, [
```
[the remark this answers is not in the document]
Comment (lukosius) [reply src/Yasumi/Provider/Japan.php]: Here the substitute holiday will always evaluate to true, so the condition is not needed. Can't find anywhere where $substituteDay be null.

https://travis-ci.org/azuyalabs/yasumi/jobs/325265019#L784

File: src/Yasumi/Provider/ChristianHolidays.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (stelgenhof) [reply src/Yasumi/Provider/ChristianHolidays.php]: Yes, it does. Maybe my memory is failing on me, haha. Anyway, agree to remove the typecast.

File: src/Yasumi/Provider/Japan.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (stelgenhof) [reply src/Yasumi/Provider/Japan.php]: Yes, you're right.

**Pull request merged**